src/app/content/routes.ts

```diff
@@ -1,15 +1,37 @@
+import omit from 'lodash/fp/omit';
 import pathToRegexp from 'path-to-regexp';
 import Loadable from 'react-loadable';
 import { Route } from '../navigation/types';
 import { getUrlRegexParams } from '../navigation/utils';
+import { assertDefined } from '../utils';
 import { SelectedResult } from './search/types';
 import { Params } from './types';
+import { findPathForParams } from './utils/urlUtils';
 
 const MATCH_UUID = '[\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12}';
+const base = '/books/:book/pages/:page';
 
-const CONTENT_PATH = '/books/:book_slug/pages/:page_slug';
-const UUID_CONTENT_PATH = `/books/:book_uuid(${MATCH_UUID})@:book_version/pages/:page_slug`;
-const VERSIONED_CONTENT_PATH = '/books/:book_slug@:book_version/pages/:page_slug';
+/*
+ * Recursively creates combinations of supplied replacements
+ * for the base parameter in an url
+ */
+
+const injectParamsToBaseUrl = (baseUrl: string, params: {[key: string]: string[]}): string[] => {
+  const paramKeys: Array<keyof typeof params> = Object.keys(params);
+  const keyToInject = paramKeys[0];
+
+  return params[keyToInject].reduce((output, value) => {
+    const injected = baseUrl.replace(new RegExp(`(?<=:)${keyToInject}`), value);
```

Comment:
does this really need the regex? can't you just do
```suggestion
    const injected = baseUrl.replace(`:${keyToInject}`, value);
```